### PR TITLE
Implement browser 'ignoreHTTPSErrors' option.

### LINKED ISSUE #84: Allow to ignore certificate errors during navigation
There should be a capability to ignore certificate errors during navigation. See https://github.com/GoogleChrome/puppeteer/issues/66 for proposal and https://github.com/cyrus-and/chrome-remote-interface/issues/183 for original discussion.

For example, we can add `ignoreSSL` navigation option:

```js
page.navigate('https://example.com', { ignoreSSL: true })
```

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch implements Browser 'ignoreHTTPSErrors' option.

Fixes #84.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -181,6 +181,7 @@ browser.newPage().then(async page => {
 
 #### new Browser([options])
 - `options` <[Object]>  Set of configurable options to set on the browser. Can have the following fields:
+  - `ignoreHTTPSErrors` <[boolean]> Whether to ignore HTTPS errors during navigation. Defaults to `false`.
```
Comment (JoelEinbinder) [inline docs/api.md]: wrong indent

File: test/test.js [hunk 2]
```diff
@@ -52,26 +52,49 @@ else
...
+    } catch (e) {
+      error = e;
+    }
+    expect(error).toBe(null);
+    expect(response.ok).toBe(true);
+    browser.close();
+  }));
+});
+
 describe('Puppeteer', function() {
```
Comment (JoelEinbinder) [inline test/test.js]: Page

File: test/test.js [hunk 3]
```diff
@@ -459,6 +482,7 @@ describe('Puppeteer', function() {
       expect(error.message).toContain('Cannot navigate to invalid URL');
     }));
     it('should fail when navigating to bad SSL', SX(async function() {
+      // Make sure events always dispatch reasonable object.
```
Comment (JoelEinbinder) [inline test/test.js]: remove or make better

Comment (aslushnikov) [reply test/test.js]: done.

File: test/test.js (hunk 2 above)
Comment (aslushnikov) [reply test/test.js]: done.

**Pull request merged**